shell/platform/darwin/macos/framework/Source/FlutterCompositor.mm

```diff
@@ -29,53 +25,16 @@
   }
 
   CGSize size = CGSizeMake(config->size.width, config->size.height);
-
-  backing_store_out->metal.struct_size = sizeof(FlutterMetalBackingStore);
-  backing_store_out->metal.texture.struct_size = sizeof(FlutterMetalTexture);
-
-  if (GetFrameStatus() != FrameStatus::kStarted) {
-    StartFrame();
-    // If the backing store is for the first layer, return the MTLTexture for the
-    // FlutterView.
-    FlutterRenderBackingStore* backingStore = [view backingStoreForSize:size];
-    backing_store_out->metal.texture.texture =
-        (__bridge FlutterMetalTextureHandle)backingStore.texture;
-  } else {
-    FlutterIOSurfaceHolder* io_surface_holder = [[FlutterIOSurfaceHolder alloc] init];
-    [io_surface_holder recreateIOSurfaceWithSize:size];
-    auto texture_descriptor =
-        [MTLTextureDescriptor texture2DDescriptorWithPixelFormat:MTLPixelFormatBGRA8Unorm
-                                                           width:size.width
-                                                          height:size.height
-                                                       mipmapped:NO];
-    texture_descriptor.usage =
-        MTLTextureUsageShaderRead | MTLTextureUsageRenderTarget | MTLTextureUsageShaderWrite;
-
-    backing_store_out->metal.texture.texture = (__bridge_retained FlutterMetalTextureHandle)
-        [mtl_device_ newTextureWithDescriptor:texture_descriptor
-                                    iosurface:[io_surface_holder ioSurface]
-                                        plane:0];
-
-    backing_store_out->metal.texture.user_data = (__bridge_retained void*)io_surface_holder;
-  }
-
+  FlutterSurface* surface = [view.surfaceManager surfaceForSize:size];
+  memset(backing_store_out, 0, sizeof(FlutterBackingStore));
+  backing_store_out->struct_size = sizeof(FlutterBackingStore);
   backing_store_out->type = kFlutterBackingStoreTypeMetal;
-  backing_store_out->metal.texture.destruction_callback = [](void* user_data) {
-    if (user_data != nullptr) {
-      CFRelease(user_data);
-    }
-  };
-
+  backing_store_out->metal.struct_size = sizeof(FlutterMetalBackingStore);
+  backing_store_out->metal.texture = surface.asFlutterMetalTexture;
   return true;
 }
 
 bool FlutterCompositor::CollectBackingStore(const FlutterBackingStore* backing_store) {
```

Comment:
Do we still need `CollectBackingStore`? Can we remove it?

Reply:
I find the only "override" of this method being `EmbedderTestCompositor`, which is not a subclass of `FlutterCompositor` despite being of a similar name.

Reply:
I removed the method. Empty callback still needs to be registered in the engine though (it's mandatory).